kratos/geometries/geometry_data.h

```diff
@@ -282,44 +281,28 @@ class GeometryData
     *        integration points.
     */
     GeometryData(GeometryDimension const *pThisGeometryDimension,
-        GeometryShapeFunctionContainer<IntegrationMethod>& ThisGeometryShapeFunctionContainer)
+        GeometryShapeFunctionContainer<IntegrationMethod> ThisGeometryShapeFunctionContainer)
```

Comment:
Why you are passing it by value? Shouldn't be const&?

Reply:
I was a bit confused with that one. I think we should discuss about this. 

The container is stored beside it. When I pass it be reference I need to copy it. Isn't const & the same? Maybe we should also consider storing the reference or even cover reference to the GeometryShapeFunctionContainer